app/main.py

```diff
@@ -1,13 +1,28 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+            "line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(err)
+                   for err in errors if err["filename"] == file_path],
+        "path": file_path,
+        "status": "passed" if not errors else "failed"
```

Comment:
You could omit using `not` by reversing the "passed" and "failed" strings placement.